src/lua/api-gateway/tracking/RequestTrackingManager.lua

```diff
@@ -229,16 +229,18 @@ local function matchVarsWithDomains(vars, domains, cache, separator)
         -- ngx.log(ngx.DEBUG, "VAR " , tostring(i))
         local variableManager = ngx.apiGateway.tracking.variableManager
         v = variableManager:getRequestVariable(vars[i], cache)
-        -- ngx.log(ngx.DEBUG, "VAL=", v)
-        d = domains[i]
-        -- ngx.log(ngx.DEBUG, "DOMAIN=", d)
-        if (d == "*") then
-            str = str .. v .. separator
```

Comment:
@andralungu what if we consider instead:
```
str = str .. tostring(v) .. separator
```

In this way the module is indeed returning all the values, even when `v` is nil, which is happening when the NGINX variable is not defined.